### PR TITLE
fix: docs and install dependencies task for new directory structure

### PR DESCRIPTION
**Pull request by ghost**
Hi, following the contributing guide I noticed a recent renaming of the `continuedev` folder to `server`. Therefore the root `install-dependencies.sh` isn't working and many links in documentation files are invalid.

### CODE REVIEW

**Comment by sestinj** [comment]
@timdahlmanns thanks so much for catching this - unfortunately it looks like since there was more than just a swapping of continuedev for server, a few of these paths don't work, for example https://github.com/continuedev/continue/tree/main/server/src/continuedev/libs/llm/llamacpp.py should be https://github.com/continuedev/continue/tree/main/server/continuedev/libs/llm/llamacpp.py

I'm happy to help out here if you're not interested in repeating this exercise, though I do notice you probably used find/replace so perhaps won't be so painful!

The old folder structure was continuedev/src/continuedev and the new structure is server/continuedev

**Comment by timdahlmanns** [comment]
@sestinj my bad, fixed it.

Great work btw, I easily added a corporate custom model by implementing a new LLM class right in the config file.

Keep it up! :+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -148,22 +148,22 @@ Everything in Continue is a "Step". The `Step` class defines 2 methods:
...
 
 Some of the most commonly used Steps are:
 
-- [`SimpleChatStep`](./continuedev/src/continuedev/plugins/steps/chat.py) - This is the default Step that is run when the user enters natural language input. It takes the user's input and runs it through the default LLM, then displays the result in the GUI.
+- [`SimpleChatStep`](./server/continuedev/plugins/steps/chat.py) - This is the default Step that is run when the user enters natural language input. It takes the user's input and runs it through the default LLM, then displays the result in the GUI.
 
-- [`EditHighlightedCodeStep`](./continuedev/src/continuedev/steps/core/core.py) - This is the Step run when a user highlights code, enters natural language, and presses CMD/CTRL+ENTER, or uses the slash command '/edit'. It opens a side-by-side diff editor, where updated code is streamed to fulfil the user's request.
+- [`EditHighlightedCodeStep`](./server/continuedev/steps/core/core.py) - This is the Step run when a user highlights code, enters natural language, and presses CMD/CTRL+ENTER, or uses the slash command '/edit'. It opens a side-by-side diff editor, where updated code is streamed to fulfil the user's request.
```
Comment (ghost) [inline CONTRIBUTING.md:157]: Ah I see there is more to do. `EditHighlightedCodeStep` moved to `server/continuedev/plugins/steps/main.py` right?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -119,13 +119,13 @@ The JetBrains extension is currently in alpha testing. Please reach out on [Disc
...
 
 ### Continue GUI Websockets Protocol
 
-On the GUI side, this is implemented in [ContinueGUIClientProtocol.ts](./extension/react-app/src/hooks/ContinueGUIClientProtocol.ts). On the server side, this is implemented in [gui.py](./continuedev/src/continuedev/server/gui.py). You can see [gui_protocol.py](./continuedev/src/continuedev/server/gui_protocol.py) or [AbstractContinueGUIClientProtocol.ts](./extension/react-app/src/hooks/AbstractContinueGUIClientProtocol.ts) for the protocol definition.
+On the GUI side, this is implemented in [ContinueGUIClientProtocol.ts](./extension/react-app/src/hooks/ContinueGUIClientProtocol.ts). On the server side, this is implemented in [gui.py](./server/continuedev/server/gui.py). You can see [gui_protocol.py](./server/continuedev/server/gui_protocol.py) or [AbstractContinueGUIClientProtocol.ts](./extension/react-app/src/hooks/AbstractContinueGUIClientProtocol.ts) for the protocol definition.
```
Comment (ghost) [inline CONTRIBUTING.md:126]: `gui_protocol.py` doesn't exist anymore

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -148,22 +148,22 @@ Everything in Continue is a "Step". The `Step` class defines 2 methods:
...
 
 ### `Policy`
 
-A `Policy` implements the method `def next(self, config: ContinueConfig, history: History) -> Step`, which decides which `Step` the `Autopilot` should run next. The default policy is defined in [policy.py](./continuedev/src/continuedev/core/policy.py) and runs `SimpleChatStep` by default, or a slash command when the input begins with '/'. It also displays a welcome message at the beginning of each session. If interested in developing agents that autonomously take longer sequences of actions in the IDE, the `Policy` class is the place to start.
+A `Policy` implements the method `def next(self, config: ContinueConfig, history: History) -> Step`, which decides which `Step` the `Autopilot` should run next. The default policy is defined in [policy.py](./server/continuedev/core/policy.py) and runs `SimpleChatStep` by default, or a slash command when the input begins with '/'. It also displays a welcome message at the beginning of each session. If interested in developing agents that autonomously take longer sequences of actions in the IDE, the `Policy` class is the place to start.
```
Comment (ghost) [inline CONTRIBUTING.md:169]: I guess this is now `server/continuedev/plugins/policies/default.py`?

File: CONTRIBUTING.md (hunk 2 above)
Comment (sestinj) [reply CONTRIBUTING.md:126]: true! I need to remove this

**Pull request merged**